src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/DecompressionHandler.cs

```diff
@@ -36,20 +36,30 @@ public DecompressionHandler(DecompressionMethods decompressionMethods, HttpMessa
         internal bool GZipEnabled => (_decompressionMethods & DecompressionMethods.GZip) != 0;
         internal bool DeflateEnabled => (_decompressionMethods & DecompressionMethods.Deflate) != 0;
         internal bool BrotliEnabled => (_decompressionMethods & DecompressionMethods.Brotli) != 0;
+        
+        private bool EncodingExists(HttpHeaderValueCollection<StringWithQualityHeaderValue> acceptEncodingHeader, string encoding)
```

Comment:
I don't know what, if anything, can be done about it in this PR, but it looks like this is going to allocate more than we do today, e.g. an enumerator for the header collection.  It's also going to force a StringWithQualityHeaderValue to be materialized for every entry in the collection, though I don't know if that was happening already.

Reply:
Would it be more expensive than the existing `Contains` check?
One thing I thought about was changing it to check all encodings in 1 loop instead of 3 loops. I didn't want to stray too far from the existing implementation, though.

Reply:
> it looks like this is going to allocate more than we do today, e.g. an enumerator for the header collection

Presumably we could fix this in `HttpHeaderValueCollection<T>` by adding a struct enumerator, right? 

Also, the common case here is that this header isn't set at all, so we could at least optimize for the case where there are no header values and return a singleton empty enumerator in that case.

> It's also going to force a StringWithQualityHeaderValue to be materialized for every entry in the collection, though I don't know if that was happening already.

Probably happening already. Definitely happening if you set the header value by constructing a StringWithQualityHeaderValue. But likely also happening anyway, even if you set the value as a string, because we always force these values to get parsed eventually before send them (currently, that is).

Reply:
> Also, the common case here is that this header isn't set at all, so we could at least optimize for the case where there are no header values and return a singleton empty enumerator in that case.

Ok, if that is indeed the common case, I'll go ahead and do that in a PR in parallel to this one.  I don't want to block this PR on adding new API.

Reply:
https://github.com/dotnet/runtime/pull/47010